sdk/storage/Azure.Storage.DataMovement/src/JobPartInternal.cs

```diff
@@ -293,7 +346,8 @@ public async virtual Task InvokeFailedArg(Exception ex)
                     _cancellationToken)).ConfigureAwait(false);
             }
             // Trigger job cancellation if the failed handler is enabled
-            await TriggerCancellation(StorageTransferStatus.CompletedWithFailedTransfers).ConfigureAwait(false);
+            await TriggerCancellationAsync().ConfigureAwait(false);
+            await CheckAndUpdateCancellationStatusAsync().ConfigureAwait(false);
```

Comment:
Just want to check that we only need to call this `CheckAndUpdateCancellationStatusAsync` on Failed? Because we expect any queued chunks to Fail since we have started a cancel?

I ask because I think if something succeeds or is skipped at the very end for some reason, it will mean the status would never be fully flipped to `Paused`?

Reply:
Actually we call `CheckAndUpdateCancellationStatusAsync` on any finished chunk. See QueueChunkToChannelAsync in JobPartInternal (I will tag you).

I also added it in other ways that the job part can end.

Though I do see your point that what if a skip happens and if QueueChunkToChannelAsync doesn't get triggered.